src/detect-rfb-secresult.c

```diff
@@ -262,38 +262,26 @@ void DetectRfbSecresultFree(DetectEngineCtx *de_ctx, void *de_ptr)
 #ifdef UNITTESTS
 /**
  * \test RfbSecresultTestParse01 is a test for a valid secresult value
- *
- *  \retval 1 on succces
- *  \retval 0 on failure
  */
 static int RfbSecresultTestParse01 (void)
 {
-    DetectRfbSecresultData *de = NULL;
-    de = DetectRfbSecresultParse("fail");
-    if (de) {
-        DetectRfbSecresultFree(NULL, de);
-        return 1;
-    }
+    DetectRfbSecresultData *de = DetectRfbSecresultParse("fail");
 
-    return 0;
+    FAIL_IF_NULL(de);
+
```

Comment:
In the case of `FAIL_IF_NULL`, we still need to call `DetectRfbSecresultFree` afterwards, because if the test doesn't fail, we need to free the resources. :)

Reply:
Thank you for the feedback.